src/OpenTelemetry.Extensions.AWS/CHANGELOG.md

```diff
@@ -15,6 +15,10 @@ Released 2024-Sep-24
 * Drop support for .NET 6 as this target is no longer supported and add .NET 8 target.
   ([#2125](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/2125))
 
+* Update `System.Text.Json` to `8.0.4` to mitigate
+  [CVE-2024-30105](https://github.com/advisories/GHSA-hh2w-p6rv-4g7w).
+  ([#2149](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2149))
```

Comment:
This is a lie for this project, not sure about others.

The reason is the `Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'"` in here...

```xml
<PackageReference Include="System.Text.Json" Version="$(SystemTextJsonPkgVer)" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
```

`net8.0` is going to stay on the framework reference version which may be 8.0.0 - 8.0.3.

What we have to do is add a direct reference to System.Text.Json when the TFM is `net8.0`.

That's why on the main repo PR I did the targets:

```xml
  <ItemGroup Condition="'$(ReferenceSystemTextJsonPackages)' == 'true' AND $(TargetFrameworksRequiringSystemTextJsonDirectReference.Contains('$(TargetFramework)'))">
    <PackageReference Include="System.Text.Json" />
  </ItemGroup>
````

Reply:
I added the reference to `System.Text.Json` for all targets in this package as it is already the case for all others like:

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/e5106b25f96815ec0c75e912bdf071f7877bd8f0/src/OpenTelemetry.Resources.Gcp/OpenTelemetry.Resources.Gcp.csproj#L18

Let me know if it is ok for you.